dev_guide/builds/build_environment.adoc

```diff
@@ -51,3 +51,26 @@ variables.
 
 Referencing container resources using `valueFrom` in build environment variables
 is not supported as the references are resolved before the container is created.
+
+[[using-secrets-as-environment-variables]]
+== Using Secrets as Environment Variables
+
+You can make key values from Secrets available as environment variables using
+the `valueFrom` syntax.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: BuildConfig
+metadata:
+  name: secret-example-bc
+spec:
+  strategy:
+    sourceStrategy:
+      env:
+      - name: MYVAL
+        valueFrom:
+          secretKeyRef:
+            key: myval
+            name: mysecret
```

Comment:
@deads2k @liggitt how much does it bother you that someone who can create a build in a namespace, can use this mechanism to access any secret in the namespace?  (because the build controller resolves this secret value on your behalf).

wondering if we need to disable the ability to make these references or at least introduce some sort of sar checking based on the builder service account's permissions.

Reply:
this is similar to the power to create a replication controller, deployment, deployment config, etc. the default roles that allow creating builds also allow creating those other workload resources.

Reply:
ok, sounds like we're fine then, thanks